### PR TITLE
Weather Sample and Upgrade Unity Version to 2022.3 LTS

### PR DESCRIPTION
**Pull request by mmgaw113**
**Sample**

Weather Sample

**Summary**

Upgrade Unity Samples to 2022.3 LTS
Weather Sample for Unity
- Rain
- Snow
- Thunder & Lightning
- Sound FX

**Tests**

Windows

**ArcGIS Maps SDK Version**

Unity 2022.3 LTS

### CODE REVIEW

**Review by avr757 — CHANGES_REQUESTED** [review]
Looks really good! The stereo sound effects are amazing! Very realistic. Great job!

Few very minor changes on the UI,
1. Outfield -> City, and then the instruction description needs to be updated
image
2. For some cities (for example, Alaminos Canyon), when you click from the dropdown, it just shows a comma on the top menu.
image
3. Please also make sure the sample viewer menu is updated :)
image

One optional suggestion, for snowy weathers, I think it would be nice to also add a sound effect, maybe something similar to this https://www.youtube.com/watch?v=7BrIJrjxVxA ?

**Comment by mmgaw113** [comment]
> Looks really good! The stereo sound effects are amazing! Very realistic. Great job!
> 
> Few very minor changes on the UI,
> 
> 1. Outfield -> Select a city.., and then the instruction description needs to be updated
>    image
> 2. For some cities (for example, Alaminos Canyon), when you click from the dropdown, it just shows a comma on the top menu.
>    image
> 3. Please also make sure the sample viewer menu is updated :)
>    image
> 
> One optional suggestion, for snowy weathers, I think it would be nice to also add a sound effect, maybe something similar to this https://www.youtube.com/watch?v=7BrIJrjxVxA ?

God idea on the sound. I'll have to add it to UE too!
Totally forgot to add to sample viewer and to update directions. Will do today!

File: samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherData.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        locationComponent = GetComponent<ArcGISLocationComponent>();
+        directionalLight = FindObjectOfType<Light>();
+        volume = FindObjectOfType<Volume>();
+        volumeProfile = volume.profile;
+        weatherQuery = FindObjectOfType<WeatherQuery>();
+        lightningAnim = directionalLight.GetComponent<Animation>();
+        lightningAudio = lightning.GetComponent<AudioSource>();
+    }
+
+    public float ConvertTemp(float Temp, bool IsOn)
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherData.cs:51]:
```suggestion
    public float ConvertTemp(float Temp, bool IsCelcius)
```

**Review by avr757 — APPROVED** [review]
Thanks for fixing the menu! and nice blizzard sounds 👍 and dont forget to add weather sample to the sample viewer menu : P

It seems that this time when I clicked on Alaminos Canyon , its showing a different city.  Since it only happens to 1 or 2 cities, it shouldn't be a big deal but just wanted to make a quick note,

image

Overall the ui/ux part looks good! Aftering adding it to the sample viewer menu it's good to go!

File: samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherData.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    public void SetSky()
+    {
+        if (volumeProfile.TryGet(out vClouds))
+        {
+            if (skyCondition.ToLower().Contains("overcast"))
+            {
+                vClouds.cloudPreset = VolumetricClouds.CloudPresets.Overcast;
+                directionalLight.color = new Color(0.1803922f, 0.1803922f, 0.1803922f, 1.0f);
+            }
+            else if (skyCondition.ToLower().Contains("Cloud"))
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherData.cs:103]:
```suggestion
            else if (skyCondition.ToLower().Contains("cloud"))
```

File: samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherQuery.cs [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+            {
+                dropDownAnim.Play("DropDownArrow");
+            }
+        });
+
+    }
+
+    private void Update()
+    {
+        arcGISCamera.gameObject.GetComponent<ArcGISCameraControllerComponent>().enabled = !MouseOverUI();
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherQuery.cs:248]: Is there a decent way to do this with an event?

File: samples_project/ProjectSettings/ArcGISProjectSettings.asset [hunk 4]
```diff
@@ -12,4 +12,4 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: 440292c6b7b936b42b794116ef2759e0, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
-  apiKey: 
+  apiKey: AAPK1733da5491544f05ae71518d08e30ea5RW7FclTJA-x6-LAshjwcCYEUVMwB41TVhmd6_BLXpj5xcgDkYuzACtVxhwpGl-oZ
```
Comment (ZackAllen) [inline samples_project/ProjectSettings/ArcGISProjectSettings.asset:15]: API key included

File: samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherQuery.cs (hunk 3 above)
Comment (mmgaw113) [reply samples_project/Assets/SampleViewer/Samples/WeatherQuery/Scripts/WeatherQuery.cs:248]: Not without creating a new script that gets attached to the UI to check on enter. This is how we do it in all the samples the require this, which is most of them.

**Review by ZackAllen — APPROVED** [review]

**Pull request merged**